src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java

```diff
@@ -163,6 +163,17 @@ public JenkinsOpenTelemetryPluginConfiguration() {
         load();
     }
 
+    @Override
+    public void load() {
+        super.load();
+        if (currentOpenTelemetryConfiguration != null) {
+            // After reloading the XML configuration, we need to reconfigure the OTel SDK, otherwise the fields here
+            // may be out of sync with the SDK. We only do this as long as `configureOpenTelemetrySdk` has run at least
+            // once so that the first configuration happens during startup via `@Initializer` after applying CasC.
+            configureOpenTelemetrySdk();
+        }
```

Comment:
This is unusual, and most plugins do not need to edit their `load` methods. In most plugins, their `GlobalConfiguration` object iself is directly the source of truth. Editing that object will directly modify all logic based on the configuration. OTel is not like that: this configuration has to be applied to the underlying SDK to take effect, so the standard load logic that just overwrites the in-memory state is not good enough. As long as you configure things through the UI this doesn't matter for OTel due to the custom logic in `configure`, but for CBCI we need similar logic in `load`.